gui/account_events.go

```diff
@@ -82,12 +82,31 @@ func (u *gtkUI) handleMessageEvent(ev events.Message) {
 	timestamp := ev.When
 	encrypted := ev.Encrypted
 	message := ev.Body
+	fromSubscribedPeer := false
+
+	sender := ev.From.NoResource().String()
 
 	p, ok := u.getPeer(account, ev.From.NoResource())
 	if ok {
+		if p.Subscription != "" {
+			fromSubscribedPeer = true
+		}
```

Comment:
This will not do the right thing - the Subscription field can be `from`, `to`, or `both` - and the peer is in charge of part of that.